src/Avalonia.Base/Styling/StyleChildren.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.ObjectModel;
+using Avalonia.Controls;
+
+namespace Avalonia.Styling
+{
+    internal class StyleChildren : Collection<IStyle>
```

Comment:
It seems to do the same as AvaloniaList with ForEachItem extension. Basically, the way as have merged dictionaries support - https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Controls/ResourceDictionary.cs#L57.

I suppose custom type makes more sense, if we don't need it to have change notifications

Reply:
I suppose custom type makes more sense, if we don't need it to have collection change notifications. Any style insertions/removal will trigger OwnerChanged event.

Reply:
Yeah it was done for performance, basically. Saves an event subscription per collection, and an event invocation per change.